src/compiler/checker.ts

```diff
@@ -21440,10 +21515,19 @@ namespace ts {
             }
             if (node.operand.kind === SyntaxKind.NumericLiteral) {
                 if (node.operator === SyntaxKind.MinusToken) {
-                    return getFreshTypeOfLiteralType(getLiteralType(-(<LiteralExpression>node.operand).text));
+                    const value = getNumericLiteralValue(node.operand as NumericLiteral);
+                    return getFreshTypeOfLiteralType(getLiteralType(
+                        // Keep number as number, bigint as bigint
+                        typeof value === "number"
+                            ? -value
+                            : { negative: !value.negative, base10Value: value.base10Value }
+                    ));
                 }
                 else if (node.operator === SyntaxKind.PlusToken) {
-                    return getFreshTypeOfLiteralType(getLiteralType(+(<LiteralExpression>node.operand).text));
+                    if (isTypeAssignableToKind(operandType, TypeFlags.BigIntLike)) {
+                        error(node.operand, Diagnostics.Operator_0_cannot_be_applied_to_type_1, tokenToString(node.operator), typeToString(operandType));
+                    }

```

Comment:
Need to handle evaluation of bigInt as enumMember value and a test. `computeConstantValue` function